webapp/tests/Controller/Jury/ContestControllerTest.php

```diff
@@ -0,0 +1,204 @@
+        array $timeString) : void
+    {
+        $this->logIn();
+        $this->verifyPageResponse('GET', static::$baseUrl, 200);
+        $link = $this->verifyLink(static::$addButton, static::$baseUrlLong.'/add');
+        $crawler = $this->client->click($link);
+
+        $h1s = $crawler->filter('h1')->extract(array('_text'));
+	    $this->assertEquals('Add contest', $h1s[0]);
+
+        $this->client->submitForm('Save', [
+            'contest[shortname]'            => hash('md5',$contest),
+            'contest[name]'                 => $contest,
+            'contest[activatetimeString]'   => $timeString[0],
+            'contest[starttimeString]'      => $timeString[1],
+            'contest[freezetimeString]'     => $timeString[2],
+            'contest[endtimeString]'        => $timeString[3],
+            'contest[unfreezetimeString]'   => $timeString[4],
+            'contest[deactivatetimeString]' => $timeString[5],
+            'contest[starttimeEnabled]'     => $radioValues[0],
+            'contest[processBalloons]'      => $radioValues[1],
+            'contest[public]'               => $radioValues[2],
+            'contest[openToAllTeams]'       => $radioValues[3],
+            'contest[enabled]'              => $radioValues[4],
+        ]);
+        foreach(['jury','admin'] as $role)
+        {
+            $this->testContestsOverview($role, 200, $contest);
+        }
+    }
+
+    public function getTimes () : array
+    {
+        $res = [[
+            "2020-01-01 09:00:00 Europe/Amsterdam",
+            "2021-01-01 09:00:00 Europe/Amsterdam",
+            "2022-01-01 09:00:00 Europe/Amsterdam",
+            "2023-01-01 09:00:00 Europe/Amsterdam",
+            "2024-01-01 09:00:00 Europe/Amsterdam",
+            "2025-01-01 09:00:00 Europe/Amsterdam",
+        ]];
+        // Check times in the past and future
+        foreach([-1,1] as $Yi)
+        {
+            $row = [];
+            foreach(['01','05','09','13','17','21'] as $H)
+            {
+                $row[] = sprintf('%s-06-01 %s:00:00 Europe/Amsterdam',
+                    date('Y') - $Yi, $H);
+            }
+            $res[] = $row;
+        }
+        for($i=-6; $i<2; $i++){
+            $row = [];
+            for($j=0; $j<6; $j++){
+                $row[] = sprintf('%s-06-01 01:00:00 Europe/Amsterdam',
+                        date('Y') + $i + $j );
+            }
+            $res[] = $row;
+        }
+        $row = ['-10:15:30.333333'];
+        $row[] = sprintf('%s-06-01 01:00:00 Europe/Amsterdam',
+            date('Y') );
+        $res[] = array_merge($row, ['+1:00','+13:00:30','+14:00','+80:00']);
+        return $res;
+    }
+
+    /**
+     * Data provider used to test if we can add contests
+     * - the to be created contest
+     * - radio button boolean values (Bit string)
+     */
+    public function provideAddContest () : \Generator
+    {
+        //* - The different times of the contest
+        //TODO: Try things like: "admin'--"
```

Comment:
I wonder if we should really test these sql injection things. This is one of the reasons we use a framework and not direct queries; the framework takes care of it.

Reply:
I agree with you that SQL injection should not happen (and if it does... upstream has to fix it) but they are interesting edge cases which are also teamnames.... but as in code it already fails its a TODO which might actually be removed in the end.

Atleast in my contest I always had people who tried DROP table... and especially in the beginning I was slightly afraid of having to enter these.. (never worked ofcourse)

I'll just remove the TODO for now.